addons/godot-xr-tools/functions/Function_Flight_movement.gd

```diff
@@ -0,0 +1,224 @@
+tool
+class_name Function_FlightMovement
+extends MovementProvider
+
+
+##
+## Movement Provider for Flying
+##
+## @desc:
+##     This script provides flying movement for the player. The control parameters
+##     are intended to support a wide variety of flight mechanics.
+##
+##     Pitch and Bearing input devices are selected which produce a "forwards"
+##     reference frame. The player controls (forwards/backwards and 
+##     left/right) are applied in relation to this reference frame.
+##
+##     The Speed Scale and Traction parameters allow primitive flight where
+##     the player is in direct control of their speed (in the reference frame).
+##     This produces an effect described as the "Mary Poppins Flying Umbrella".
+##
+##     The Acceleration, Drag, and Guidance parameters allow for slightly more
+##     realisitic flying where the player can accelerate in their reference
+##     frame. The drag is applied against the global reference and can be used
+##     to construct a terminal velocity.
+##
+##     The Guidance property attempts to lerp the players velocity into flight
+##     forwards direction as if the player had guide-fins or wings.
+##
+##     The Exclusive property specifies whether flight is exclusive (no further
+##     physics effects after flying) or whether additional effects such as 
+##     the default player gravity are applied.
+##
+
+
+# enum our buttons, should find a way to put this more central
+enum Buttons {
```

Comment:
@Malcolmnixon probably not for this PR but maybe a separate one, we should look into moving this into our util script so these enums are just available